substrate/client/db/src/lib.rs

```diff
@@ -304,30 +309,43 @@ impl state_machine::Backend for DbState {
 
 /// In-memory backend. Keeps all states and blocks in memory. Useful for testing.
 pub struct Backend {
-	db: Arc<Database>,
+	db: Arc<KeyValueDB>,
 	blockchain: BlockchainDb,
 	old_states: RwLock<HashMap<BlockKey, state_machine::backend::InMemory>>,
 }
 
 impl Backend {
-	/// Create a new instance of in-mem backend.
+	/// Create a new instance of database backend.
 	pub fn new(config: &DatabaseSettings) -> Result<Backend, client::error::Error> {
-		let mut db_config = DatabaseConfig::with_columns(columns::NUM_COLUMNS);
+		let mut db_config = DatabaseConfig::with_columns(Some(columns::NUM_COLUMNS));
 		db_config.memory_budget = config.cache_size;
 		db_config.wal = true;
 		let path = config.path.to_str().ok_or_else(|| client::error::ErrorKind::Backend("Invalid database path".into()))?;
 		let db = Arc::new(Database::open(&db_config, &path).map_err(db_err)?);
+
+		Backend::from_kvdb(db as Arc<_>)
+	}
+
+	#[cfg(test)]
+	fn new_test() -> Backend {
+		let db = Arc::new(::kvdb_memorydb::create(columns::NUM_COLUMNS));
+
+		Backend::from_kvdb(db as Arc<_>).expect("failed to create test-db")
+	}
+
+	fn from_kvdb(db: Arc<KeyValueDB>) -> Result<Backend, client::error::Error> {
 		let blockchain = BlockchainDb::new(db.clone())?;
 
 		//load latest state
 		let mut state = state_machine::backend::InMemory::new();
 		let mut old_states = HashMap::new();
-	 	if let Some(iter) = db.iter(columns::STATE).map(|iter| iter.map(|(k, v)| (k.to_vec(), Some(v.to_vec())))) {
+
+		{
+	 		let iter = db.iter(columns::STATE).map(|(k, v)| (k.to_vec(), Some(v.to_vec())));
 			state.commit(iter);
 			old_states.insert(number_to_db_key(blockchain.meta.read().best_number), state);
 		}
 
-		debug!("DB Opened at {}", path);
```

Comment:
What's wrong with this log?

Reply:
could add it back, but it would need to be in the wrapping function, not this one